### PR TITLE
Security annotation added, to make methods only be callable by a admin user

### PR DESCRIPTION
**Pull request by simon-ittmann**

### CODE REVIEW

File: src/main/resources/application-non-sec.properties [hunk 1]
```diff
@@ -1,3 +1,5 @@
-# application constants
-## will be used as default user subject if the security is disabled
-app.non-sec-oauth2-subject=non-sec-oauth2-subject
\ No newline at end of file
+# will be used as default user subject if the security is disabled
+app.non-sec-user-oauth2-subject=non-sec-user-oauth2-subject
+
+# weather the default non-sec user should be admin
```
Comment (ptolstoi) [inline src/main/resources/application-non-sec.properties:4]:
```suggestion
# whether the default non-sec user should be admin
```

File: src/test/kotlin/de/byteleaf/companyon/auth/oauth/JwtMock.kt [hunk 2]
```diff
@@ -1,10 +1,6 @@
-package de.byteleaf.companyon.security.converter
+package de.byteleaf.companyon.auth.oauth
 import org.springframework.security.oauth2.jwt.Jwt
 
-/**
```
Comment (ptolstoi) [inline src/test/kotlin/de/byteleaf/companyon/auth/oauth/JwtMock.kt:4]: was this change intended?

File: src/test/kotlin/de/byteleaf/companyon/auth/oauth/OAuth2JwtAuthenticationConverterTest.kt [hunk 3]
```diff
@@ -26,7 +26,6 @@ import org.springframework.test.context.ActiveProfiles
 class OAuth2JwtAuthenticationConverterTest {
 
 
-    private val TEST_SUBJECT = "test-subject"
```
Comment (ptolstoi) [inline src/test/kotlin/de/byteleaf/companyon/auth/oauth/OAuth2JwtAuthenticationConverterTest.kt:29]: was this change intended?

File: src/main/kotlin/de/byteleaf/companyon/auth/configuration/MethodSecurityConfiguration.kt [hunk 4]
```diff
@@ -0,0 +1,9 @@
+package de.byteleaf.companyon.auth.configuration
+
+import org.springframework.context.annotation.Configuration
+import org.springframework.security.config.annotation.method.configuration.EnableGlobalMethodSecurity
+import org.springframework.security.config.annotation.method.configuration.GlobalMethodSecurityConfiguration
+
+@Configuration
+@EnableGlobalMethodSecurity(prePostEnabled = true, securedEnabled = true)
+class MethodSecurityConfiguration : GlobalMethodSecurityConfiguration()
```
Comment (ptolstoi) [inline src/main/kotlin/de/byteleaf/companyon/auth/configuration/MethodSecurityConfiguration.kt:9]: I can't start the application anymore because I'm getting an error here:

```
The bean 'methodSecurityInterceptor', defined in class path resource [org/springframework/security/config/annotation/method/configuration/GlobalMethodSecurityConfiguration.class], could not be registered. A bean with that name has already been defined in class path resource [de/byteleaf/companyon/auth/configuration/MethodSecurityConfiguration.class] and overriding is disabled.
```

**Review by ptolstoi — CHANGES_REQUESTED** [review]

File: src/main/kotlin/de/byteleaf/companyon/user/dto/User.kt [hunk 5]
```diff
@@ -1,15 +1,22 @@
...
     val firstName: String,
     val lastName: String,
     val email: String,
+    val admin: Boolean = false,
     val signature: FileMeta?,
     val avatar: FileMeta?
-) : BaseDTO()
+) : BaseDTO() {
+
+    fun getRoles(): List<GrantedAuthority> =
```
[the remark this answers is not in the document]
Comment (simon-ittmann) [reply src/main/kotlin/de/byteleaf/companyon/user/dto/User.kt:20]: https://www.baeldung.com/spring-security-granted-authority-vs-role

File: src/main/resources/application-non-sec.properties (hunk 1 above)
Comment (simon-ittmann) [reply src/main/resources/application-non-sec.properties:4]: merci :)

File: src/test/kotlin/de/byteleaf/companyon/auth/oauth/JwtMock.kt (hunk 2 above)
Comment (simon-ittmann) [reply src/test/kotlin/de/byteleaf/companyon/auth/oauth/JwtMock.kt:4]: yes I renamed the package from securtiy back to auth and this happend during the merge

File: src/main/kotlin/de/byteleaf/companyon/auth/configuration/MethodSecurityConfiguration.kt (hunk 4 above)
Comment (simon-ittmann) [reply src/main/kotlin/de/byteleaf/companyon/auth/configuration/MethodSecurityConfiguration.kt:9]: yes sorry, thanks for finding ;)

File: src/test/kotlin/de/byteleaf/companyon/auth/oauth/OAuth2JwtAuthenticationConverterTest.kt (hunk 3 above)
Comment (simon-ittmann) [reply src/test/kotlin/de/byteleaf/companyon/auth/oauth/OAuth2JwtAuthenticationConverterTest.kt:29]: yes, bad merge from myside

**Review by ptolstoi — APPROVED** [review]
